README.md

```diff
@@ -1,18 +1,334 @@
-## Update/Delete Nodes With DevTools
+A variable is a container in which we can store values for later retrieval.
+
+Imagine a box that can hold any type of data: a `number`, `string`, `boolean`,
+`object` — even an `undefined`. We take some data that we want to store, place
+it inside the box, and hand the box off to the JavaScript engine, which stores
+it in memory. All done! Our data is safely cached until we need to access it
+again.
+
+But wait! When we ask for the data back, how will the JavaScript engine know
+_which_ box to retrieve? We need to assign a name to our variable — a label
+for our box — so that we can tell the engine exactly which piece of stored
+data we want to access.
+
+## Name Variables in JavaScript
+
+Variable names in JavaScript can sometimes be complicated, but if you follow
+these three rules you'll be fine:
+
+- Start every variable name with a lowercase letter. Variable names starting
+with a number are not valid.
+- Don't use spaces — `camelCaseYourVariableNames` (see the camel humps?) instead
+of `snake_casing_them` (like the underscore is a snake that swallowed the words).
+- Don't use JavaScript [reserved words][reserved words] or [future reserved words][future reserved words].
+
+It's important to note that case matters, so `javaScript`, `javascript`,
+`JavaScript`, and `JAVASCRIPT` are four different variables.
+
+## Initialize Variables in JavaScript
+
+The `var` reserved word is the classic way to declare a variable. It's been
+around since the inception of JavaScript, and it's what you will encounter
+in any pre-ES2015 code.
+
+Creating new variables in JavaScript is really a two-step process. First,
+we declare the variable:
+
+```js
+var pi;
+//=> undefined
+```
+
+The JavaScript engine sets aside a chunk of memory to store the declared
+variable. Then, we assign a value to that variable:
+
+```js
+pi = 3.14159;
+//=> 3.14159
+```
+
+We can package both of the initialization steps — declaration and assignment
+— in a single line of code:
+
+```js
+var pi = 3.14159;
+//=> undefined
+```
+
+
+### Multi-line Variable Assignment
+
+Let's say I needed to declare and define multiple variables. It feels like a
+lot to have to repeat `var` over and over again. JavaScript allows us to do
+multi-line variable assignment to alleviate this pain. Every variable must be
+separated with a comma, and the entire line must end with a semicolon.
+
+Let's condense the below code into one line:
+
+```javascript
+var a = 5;
+var b = 2;
+var c = 3;
+var d = {};
+var e = [];
+```
+
+The above is equivalent to:
+
+```javascript
+var a = 5,
+    b = 2,
+    c = 3,
+    d = {},
+    e = [];
+```
+
+which can be converted to:
+
+```javascript
+var a = 5, b = 2, c = 3, d = {}, e = [];
+```
+
+### Retrieving Variables
+
+To retrieve a declared variable, refer to its name:
+```js
+pi;
+//=> 3.14159
+```
+
+### Reassigning Variable Value
+
+Changing the value of a variable in JavaScript works just in the same way
+as it does in Ruby:
```

Comment:
We should remove all comparisons to Ruby, since we're moving to a modular curriculum and this piece might one day be used without Ruby pieces.

Reply:
True